File: 01/day01.f90
program day01
    use aoc

    implicit none

    integer :: n_lines
    integer, dimension(:), allocatable :: input_data, differences

    integer :: i, v, n_inc

    call count_lines('input.txt', n_lines)

    allocate(input_data(n_lines))

    open(7, file='input.txt')
    do i = 1, n_lines
        read(7, fmt=*) input_data(i)
    enddo
    close(7)

!   part 1

    allocate(differences(n_lines - 1))

    differences = input_data(2:n_lines) - input_data(1:n_lines-1)

    n_inc = 0

    do i = 1,n_lines-1
        if (differences(i) > 0) n_inc = n_inc + 1
    enddo

    write(*,*) n_inc

    deallocate(differences)
    
!   part 2

    allocate(differences(n_lines - 3))
    differences = input_data(4:n_lines) - input_data(1:n_lines-1)

    n_inc = 0

    do i = 1,n_lines-3
        if (differences(i) > 0) n_inc = n_inc + 1
    enddo

    write(*,*) n_inc

!
    deallocate(input_data)

end program

File: 01/day01.py
import numpy as np

with open('input.txt', mode='r') as f:
    data = np.array([int(l.strip()) for l in f.readlines()])

# part 1
differences = data[1:] - data[0:-1]
num_inc = sum(differences > 0)
print(num_inc)

# part 2
sums_3 = np.array([sum(data[i:i+3]) for i in range(len(data)-2)])
differences = sums_3[1:] - sums_3[0:-1]
num_inc = sum(differences > 0)
print(num_inc)